src/cascadia/TerminalSettingsModel/TerminalSettings.cpp

```diff
@@ -124,6 +124,21 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
             {
                 defaultSettings.StartingTitle(newTerminalArgs.TabTitle());
             }
+            else
+            {
+                // There was no title, and no profile from which to infer the title.
+                // Per GH#6776, promote the first component of the command line to the title.
+                // This will ensure that the tab we spawn has a name (since it didn't get one from its profile!)
+                if (newTerminalArgs.Profile().empty() && !newTerminalArgs.Commandline().empty())
+                {
+                    const std::wstring_view commandLine{ newTerminalArgs.Commandline() };
+                    const auto start{ til::at(commandLine, 0) == L'"' ? 1 : 0 };
+                    const auto terminator{ commandLine.find_first_of(start ? L'"' : L' ', start) }; // look past the first character if it starts with "
+                    // We have to take a copy here; winrt::param::hstring requires a null-terminated string

```

Comment:
this is rough but it works. doing a substr with `npos-x` later is safe (confirmed by docs)

Reply:
if it yields a title of _nothing_, we already have a weird commandline. If the title is too long, _oh well_

Reply:
Are you sure about this? `winrt::param::hstring` has an explicit `std::wstring_view` constructor and appears to work just fine with it.

Reply:
Dead sure. This blog post just came out today to reinforce: https://devblogs.microsoft.com/oldnewthing/20210823-00/?p=105598

Reply:
![shocked](https://media.giphy.com/media/wHE6Dd6RCVHQfjK5dy/giphy-downsized.gif?cid=790b76110ce55b3c0124956552719983253c9138acbb626b&rid=giphy-downsized.gif&ct=g)

Reply:
Uh oh.